src/ngraph/runtime/intelgpu/intelgpu_op_batchnorm.cpp

```diff
@@ -0,0 +1,149 @@
+{
+    throw invalid_argument(
+        "intelgpu::get_batch_norm_mean() Calculation matrix mean is not yet supported.");
+}
+
+string get_batch_norm_variance(cldnn::topology& topology,
+                               const string& input_name,
+                               const string& mean_name)
+{
+    throw invalid_argument(
+        "intelgpu::get_batch_norm_variance() Calculation matrix variance is not yet supported.");
+}
+
+void runtime::intelgpu::do_batch_norm_operation(cldnn::topology& topology,
+                                                const string& output_name,
+                                                double eps,
+                                                const string& input_name,
+                                                const Shape& input_shape,
+                                                const string& gamma_name,
+                                                const string& beta_name,
+                                                const string& mean_name_inp,
+                                                const string& variance_name_inp)
+{
+    vector<pair<cldnn::primitive_id, cldnn::tensor>> split_offsets;
+    vector<pair<cldnn::primitive_id, cldnn::tensor>> vec_offsets;
+    vector<cldnn::primitive_id> dim_set;
+
+    if (input_shape.size() < 2 || input_shape.size() > 4)
+    {
+        throw invalid_argument("intelgpu::do_batch_norm_operation() wrong input shape.");
+    }
+
+    // According to the documentation, input data channel is always being axis 1
+    // Assumed the second dimension from the left. Example {0, 1, 0, 0} or {0, 1}
+    // Also, input data must be at least 2D array
+    const size_t shape_channel = 1;
+    const size_t cldnn_channel = 4 - input_shape.size() + shape_channel;
+
+    const size_t splitted_arr_count = input_shape.at(shape_channel);
+    for (size_t i = 0; i < splitted_arr_count; ++i)
+    {
+        const string str_i = to_string(i);
+        const cldnn::tensor vec_offset(0, 0, i, 0);
+        vec_offsets.push_back(pair<cldnn::primitive_id, cldnn::tensor>(str_i, vec_offset));
+
+        vector<cldnn::tensor::value_type> offset({0, 0, 0, 0}); // No action by default
+        offset.at(cldnn_channel) = i;
+
+        cout << "Splitted to " << i << " with " << vector_to_string(offset) << "\n";
+        const cldnn::tensor input_offset(offset.at(0), offset.at(1), offset.at(3), offset.at(2));
+        split_offsets.push_back(pair<cldnn::primitive_id, cldnn::tensor>(str_i, input_offset));
+    }
+
+    string mean_name = mean_name_inp;
+    if (mean_name_inp.empty())
+    {
+        mean_name = get_batch_norm_mean(topology, input_name);
+    }
+
+    string variance_name = variance_name_inp;
+    if (variance_name_inp.empty())
+    {
+        variance_name = get_batch_norm_variance(topology, input_name, mean_name);
+    }
+
+    const string input_splitted_name = do_matrix_split(topology, input_name, split_offsets);
```

Comment:
Can we change the part of the name `splitted` to `split`, which is the more correct past tense of split. This applies to all variables with `splitted` in the name.